tests/integration/sharding_tests/test_sharding_backups.py

```diff
@@ -200,3 +216,89 @@ async def test_rotate_backup_password(ops_test: OpsTest) -> None:
                 assert backups == 2
     except RetryError:
         assert backups == 2, "Backup not created after password rotation."
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_restore_backup(ops_test: OpsTest, add_writes_to_db) -> None:
+    """Tests that sharded Charmed MongoDB cluster supports restores."""
+    # count total writes
+    cluster_writes = await writes_helpers.get_cluster_writes_count(
+        ops_test, shard_app_names=SHARD_APPS
+    )
+    assert cluster_writes["total_writes"] > 0, "no writes to backup"
+
+    leader_unit = await backup_helpers.get_leader_unit(
+        ops_test, db_app_name=CONFIG_SERVER_APP_NAME
+    )
+    prev_backups = await backup_helpers.count_logical_backups(leader_unit)
+    await ops_test.model.wait_for_idle(
+        apps=[CONFIG_SERVER_APP_NAME], status="active", idle_period=20
+    ),
+    action = await leader_unit.run_action(action_name="create-backup")
+    first_backup = await action.wait()
+    assert first_backup.status == "completed", "First backup not started."
+
+    # verify that backup was made on the bucket
+    try:
+        for attempt in Retrying(stop=stop_after_delay(4), wait=wait_fixed(5)):
+            with attempt:
+                backups = await backup_helpers.count_logical_backups(leader_unit)
+                assert backups == prev_backups + 1, "Backup not created."
+    except RetryError:
+        assert backups == prev_backups + 1, "Backup not created."
```

Comment:
is this needed?

Reply:
Yes - the reason is I don't want to just assert failure when the `RetryError` occurs, I want to know exactly what the error was and what the count was, what was the expected count, and a helpful message.

Reply:
Yes, you essentially want to propagate the assertion failure instead of an obscure `RetryError`. 
You can achieve this by adding `reraise=True` in the `Retrying` context, and you can drop the `except RetryError`